### PR TITLE
Add basic default setup for Tailwind

### PR DESCRIPTION
**Pull request by iHiD**
We're going to try using Tailwind for some CSS stuff. This is the basic setup for it. Not sure if it'll stay long term but let's see 🙂 

This also adds some very basic styling for the track component as an example. 

Naming of classes/tailwind variables are just placeholders for now - no need to comment on those.

Thoughts welcome 🙂

### CODE REVIEW

File: postcss.config.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module.exports = {
   plugins: [
     require('postcss-import'),
+    require('tailwindcss'),
+    //require('autoprefixer'),
```
Comment (iHiD) [inline postcss.config.js]: Everything says we should have this uncommented but I'm not sure why yet. I'm leaving it commented for now so if something doesn't work later I can remember to try uncommenting this, but I'll remove it if we don't use it once we're going.

File: tailwind.config.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    // removeDeprecatedGapUtilities: true,
+    // purgeLayersByDefault: true,
+  },
+  purge: [],
+  theme: {
+    extend: {},
+  },
+  variants: {},
+  plugins: [],
+}
```
Comment (iHiD) [inline tailwind.config.js:12]: This is the default generated config.

File: postcss.config.js (hunk 1 above)
Comment (neenjaw) [reply postcss.config.js]: From what I understand, tailwind doesn't need autoprefixer to work, but it does create css rules that may require vendor prefixes in some browsers. [This is the best I can find right now for proof of this](https://github.com/tailwindlabs/tailwindcss/issues/291)

Comment (iHiD) [reply postcss.config.js]: @neenjaw Thanks 🙂  Turns out that `postcss-preset-env` already has support for auto-prefixing so this isn't required at all.

File: app/javascript/css/application.css [hunk 3]
```diff
@@ -14,4 +14,23 @@
...
+            @apply font-bold;
+        }
+    }
+}
+
+input {
+    &::placeholder {
+        color: #123456;
+    }
+}
```
Comment (iHiD) [inline app/javascript/css/application.css]: I've done this manually to test that auto-prefixes get added (they do). Any thoughts on how we could write a regression test to ensure this works (ie, we don't accidently lose the prefixing later down the line). (cc @SleeplessByte too)

Comment (iHiD) [inline app/javascript/css/application.css]: This in general is just me checking that things are working.

File: app/views/pages/index.html.haml [hunk 4]
```diff
@@ -1,4 +1,15 @@
...
+.text-purple-700 Some purple text
+.hello 
+  Hello, 
+  .world World!
+
+.tw-text-green-400 
+  Hello
+  .tw-inline-block.md:tw-font-bold World!
+
+%input.tw-border-solid.tw-border-4.tw-border-green-500{placeholder: "foobar"}
```
Comment (iHiD) [inline app/views/pages/index.html.haml]: This is also just me POC'ing.

File: app/javascript/css/application.css (hunk 3 above)
Comment (neenjaw) [reply app/javascript/css/application.css]: I safari still uses a lot of vendor tags, you could write a test on a flex-box enable page to look for that css vendor prefix.

- e.g. http://shouldiprefix.com/#flexbox

Comment (iHiD) [reply app/javascript/css/application.css]: I'm going to move this to a separate issue: https://github.com/exercism/v3-website/issues/123

File: tailwind.config.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+module.exports = {
+  future: {
+    // removeDeprecatedGapUtilities: true,
+    // purgeLayersByDefault: true,
```
Comment (iHiD) [inline tailwind.config.js:4]: These are added by default - I've not checked what they mean yet.

File: app/javascript/css/application.css [hunk 6]
```diff
@@ -14,4 +14,10 @@
  *= require_self
  */
 
-.hello { color: green; }
+@import "tailwindcss/base";
+@import "tailwindcss/components";
+@import "tailwindcss/utilities";
+
+body {
+    @apply font-body bg-bgGray antialiased;
```
Comment (iHiD) [inline app/javascript/css/application.css:22]: Re antialiased: https://github.com/exercism/v3-website/issues/124

File: app/javascript/css/components/track.css [hunk 7]
```diff
@@ -0,0 +1,17 @@
+@import "../styles";
```
Comment (kntsoriano) [inline app/javascript/css/components/track.css:1]: What does `&` mean here?

**Review by kntsoriano — APPROVED** [review]
Looks good! Does this work outside React? Just thinking we might need to enforce consistency between those two (component styles vs non-component styles) if ever it's needed.

**Review by ErikSchierboom — APPROVED** [review]
Looks good!

File: app/javascript/css/components/track.css (hunk 7 above)
Comment (iHiD) [reply app/javascript/css/components/track.css:1]: `&` is the CSS3 nesting operator. Whereas in sass it's optional (and normally not added), in CSS3 it's compulsory, so every nested style has to have it.

**Comment by iHiD** [comment]
> Does this work outside React?

Yeah - this isn't react at all. It's postcss (which is basically babel for css - so you write css3 and it downgrades to css2) and then tailwind as a library on it. All our CSS will go through this (rather than the asset pipeline) regardless of whether it's for JS components or normal things.

**Comment by taiyab** [comment]
Big fan of Tailwind 🚀

**Pull request merged**